src/api/users.py

```diff
@@ -85,3 +68,89 @@ def delete_user(user_id: int):
             {"user_id": user_id},
         )
     return "OK"
+
+
+@router.get("/{user_id}/inventory")
+def get_inventory(user_id: int):
+    """
+    Provides the username, amount of gold, and treats in the inventory of the user
+    with the given id.
+    """
+    try:
+        with db.engine.begin() as connection:
+            inventory = connection.execute(
+                sqlalchemy.text(
+                    """
+                    SELECT
+                        u.username,
+                        g.gold,
+                        i.treat_sku,
+                        i.quantity
+                    FROM users u
+                    LEFT JOIN user_inventory_view i
+                        ON i.user_id = u.id
+                        AND i.quantity > 0
+                    LEFT JOIN gold_view g
+                        ON g.user_id = u.id
+                    WHERE u.id = :id
+                """
+                ),
+                {"id": user_id},
+            ).mappings()
+
+        treat_list = []
+        username = None
+        gold = None
+
+        for item in inventory:
+            if username is None and gold is None:
+                username = item["username"]
+                gold = item["gold"]
+
+            if item["treat_sku"]:
+                treat_list.append(
+                    {"username": item["treat_sku"], "quantity": item["quantity"]}
+                )
+
+        if username == None or gold == None:
+            return {"success": False, "error": f"User {user_id} does not exist."}
+
+        return {"name": username, "gold": gold, "treats": treat_list}
+
+    except Exception as e:
+        print(f"An unexpected error has occurred: {e}")
+        return {"success": False, "error": str(e)}
+
+
+@router.get("/{user_id}/adoptions")
+def get_adoptions(user_id: int):
+    """
+    Provides the name and stage of each creature the user with the given id has adopted.
+    """
+    try:
+        with db.engine.begin() as connection:
+            adoptions = (
+                connection.execute(
+                    sqlalchemy.text(
+                        """
+                    SELECT 
+                        name, stage
+                    FROM user_creature_connection u
+                    LEFT JOIN creatures c ON u.creature_id = c.id
```

Comment:
why a LEFT JOIN and not a JOIN? LEFT JOIN implies that it's possible for user_creature_connection to have creatures that don't exist in the creatures table

Reply:
True, I'll fix that 👍